code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/PortsConstraintProvider.java

```diff
@@ -0,0 +1,33 @@
+package io.cattle.platform.allocator.constraint;
+
+import io.cattle.platform.allocator.dao.AllocatorDao;
+import io.cattle.platform.allocator.service.AllocationAttempt;
+import io.cattle.platform.allocator.service.AllocationLog;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.core.model.Port;
+import io.cattle.platform.object.ObjectManager;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class PortsConstraintProvider implements AllocationConstraintsProvider {
+
+    @Inject
+    AllocatorDao allocatorDao;
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Override
+    public void appendConstraints(AllocationAttempt attempt, AllocationLog log,
+            List<Constraint> constraints) {
+        Instance instance = attempt.getInstance();
+
+        List<Port> ports = objectManager.children(instance, Port.class);
```

Comment:
It can.  I'll fix this.  Thanks for finding this.  It's true things in the container can change prior to starting.